src/main/java/iroha/validation/transactions/provider/impl/util/CacheProvider.java

```diff
@@ -25,13 +24,17 @@
   public synchronized void put(Transaction transaction) {
     final String accountId = ValidationUtils.getTxAccountId(transaction);
     if (!cache.containsKey(accountId)) {
-      cache.put(accountId, new ArrayList<>());
+      cache.put(accountId, new LinkedList<>());
```

Comment:
Why LinkedList?

Reply:
Because we add to the end and take from the head only. I think it is convenient to use linked implementation